### PR TITLE
docs: Add parameter for ODBC SQL dialect

### PR DESCRIPTION
**Pull request by peasee**
## 🗣 Description

* Renames old parameters to prefix them with `odbc_`
* Adds parameter information for `sql_dialect`

## 🔨 Related Issues

* Part of [https://github.com/spiceai/spiceai/issues/1989](https://github.com/spiceai/spiceai/issues/1989)

### CODE REVIEW

File: spiceaidocs/docs/components/data-connectors/odbc.md [hunk 1]
```diff
@@ -28,11 +28,12 @@ In addition to the connection string, the following [arrow_odbc builder paramete
...
-| max_text_size           | number (bytes) | Upper limit for value buffers bound to columns with text values.                                                                                                                                 | Unset (allocates driver-reported max column size) |
-| max_binary_size         | number (bytes) | Upper limit for value buffers bound to columns with binary values.                                                                                                                               | Unset (allocates driver-reported max column size) |
-| enable_db2_length_quirk | bool           | If connecting to IBM DB2, enable a [quirk](https://docs.rs/arrow-odbc/latest/arrow_odbc/struct.Quirks.html#structfield.indicators_returned_from_bulk_fetch_are_memory_garbage) to avoid crashing.| `false`                                           |
+| odbc_sql_dialect     | string | Override what SQL dialect is used for the ODBC connection. Supports `postgresql`, `mysql`, `sqlite` or `databricks` values.   | Unset (auto-detected) |
```
Comment (phillipleblanc) [inline spiceaidocs/docs/components/data-connectors/odbc.md]:
```suggestion
| sql_dialect     | string | Override what SQL dialect is used for the ODBC connection. Supports `postgresql`, `mysql`, `sqlite` or `databricks` values.   | Unset (auto-detected) |
```

File: spiceaidocs/docs/components/data-connectors/odbc.md [hunk 2]
```diff
@@ -28,11 +28,12 @@ In addition to the connection string, the following [arrow_odbc builder paramete
...
+| odbc_max_bytes_per_batch     | number (bytes) | Upper allocation limit for transit buffer.   | `512_000_000` |
+| odbc_max_num_rows_per_batch  | number (rows)  | Upper limit for number of rows fetched for one batch. | `65536` |
+| odbc_max_text_size           | number (bytes) | Upper limit for value buffers bound to columns with text values. | Unset (allocates driver-reported max column size) |
+| odbc_max_binary_size         | number (bytes) | Upper limit for value buffers bound to columns with binary values. | Unset (allocates driver-reported max column size) |
+| odbc_enable_db2_length_quirk | bool           | If connecting to IBM DB2, enable a [quirk](https://docs.rs/arrow-odbc/latest/arrow_odbc/struct.Quirks.html#structfield.indicators_returned_from_bulk_fetch_are_memory_garbage) to avoid crashing. | `false` |
```
Comment (phillipleblanc) [inline spiceaidocs/docs/components/data-connectors/odbc.md]:
```suggestion
```

This parameter isn't supported anymore

File: spiceaidocs/docs/components/data-connectors/odbc.md [hunk 3]
```diff
@@ -42,6 +43,27 @@ datasets:
       odbc_connection_string: Driver={Foo Driver};Host=db.foo.net;Param=Value
 ```
 
+### Selecting SQL Dialect
+
+The default SQL dialect may not be supported by every ODBC connection. The `odbc_sql_dialect` allows overriding the selected SQL dialect for a specified connection.
```
Comment (phillipleblanc) [inline spiceaidocs/docs/components/data-connectors/odbc.md]:
```suggestion
The default SQL dialect may not be supported by every ODBC connection. The `sql_dialect` parameter allows overriding the selected SQL dialect for a specified connection.
```

File: spiceaidocs/docs/components/data-connectors/odbc.md [hunk 4]
```diff
@@ -42,6 +43,27 @@ datasets:
...
+
+The runtime will attempt to detect the dialect to use for a connection based on the contents of `Driver=` in the `odbc_connection_string`. The runtime will usually detect the correct SQL dialect for the following connection types:
+
+* PostgreSQL
+* MySQL
+* SQLite
+* Databricks
+
+These connection types are also the supported values for overriding dialect in `odbc_sql_dialect`. For example, overriding the dialect for your connection to a `postgresql` style dialect:
```
Comment (phillipleblanc) [inline spiceaidocs/docs/components/data-connectors/odbc.md]:
```suggestion
These connection types are also the supported values for overriding dialect in `sql_dialect`. For example, overriding the dialect for your connection to a `postgresql` style dialect:
```

File: spiceaidocs/docs/components/data-connectors/odbc.md [hunk 5]
```diff
@@ -42,6 +43,27 @@ datasets:
...
+```yaml
+datasets:
+  - from: odbc:path.to.my_dataset
+    name: my_dataset
+    params:
+      odbc_sql_dialect: postgresql
```
Comment (phillipleblanc) [inline spiceaidocs/docs/components/data-connectors/odbc.md]:
```suggestion
      sql_dialect: postgresql
```

File: spiceaidocs/docs/components/data-connectors/odbc.md (hunk 4 above)
Comment (phillipleblanc) [inline spiceaidocs/docs/components/data-connectors/odbc.md]: Can you have a list of the supported values here that are accepted by `sql_dialect`, similar to what you have above.

File: spiceaidocs/docs/components/data-connectors/odbc.md (hunk 1 above)
Comment (peasee) [reply spiceaidocs/docs/components/data-connectors/odbc.md]: The parameter is `odbc_sql_dialect`. `sql_dialect` makes no effect?

Comment (phillipleblanc) [reply spiceaidocs/docs/components/data-connectors/odbc.md]: This was changed to `sql_dialect` in https://github.com/spiceai/spiceai/pull/2028

Comment (peasee) [reply spiceaidocs/docs/components/data-connectors/odbc.md]: bug fixed in https://github.com/spiceai/spiceai/pull/2050

**Review by digadeesh — DISMISSED** [review]

**Review by digadeesh — APPROVED** [review]

**Pull request merged**